### PR TITLE
search: fix mappings for records and deposits

### PR DESCRIPTION
**Pull request by mitsosf**

### CODE REVIEW

**Comment by ppanero** [comment]
Nice! much needed cleanup, thanks! I am wondering if some cases would need both (i.e. text + keyword). For example name, if I search for `pab` I would like `pablo` to be returned. If it is keyword only `pablo` (i.e. exact match) would return a hit.

**Comment by mitsosf** [comment]
Yes, we will take a final look at the end to discuss what changes stay or go, just doing a first pass.

**Review by slint — COMMENTED** [review]
Some minor corrections for description-like fields and a special case. Check also that they're applied on the `record`  mapping.

**Comment by slint** [comment]
> For example name, if I search for `pab` I would like `pablo` to be returned. If it is keyword only `pablo` (i.e. exact match) would return a hit.

I tried to verify this and had to change again my entire internal mental model of how ES types work...  So, here's the summary:

<table>
<thead>
  <tr>
    <th>Value 👉 </th>
    <th colspan="2">"Pablo Panero"</th>
    <th colspan="2">"Panero"</th>
  </tr>
</thead>
<tbody>
  <tr>
    <td>Query 👇 | Type 👉 </td>
    <td><code>text</code></td>
    <td><code>keyword</code></td>
    <td><code>text</code></td>
    <td><code>keyword</code></td>
  </tr>
  <tr>
    <td><code>pan</code></td>
    <td>❌</td>
    <td>❌</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td><code>panero</code></td>
    <td>✅</td>
    <td>❌</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td><code>pablo</code></td>
    <td>✅</td>
    <td>❌</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td><code>pablo*</code></td>
    <td>✅</td>
    <td>✅</td>
    <td>❌</td>
    <td>❌</td>
  </tr>
  <tr>
    <td><code>pan*</code></td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
    <td>✅</td>
  </tr>
</tbody>
</table>

- Partial matches (i.e. just `pan`) don't work... Word stemming doesn't actually apply by default. Wildcards (i.e. `*`) and regex queries work though :)
- Case sensitivity doesn't matter in any case... I think it's because [the standard analyzer](https://www.elastic.co/guide/en/elasticsearch/reference/current/analysis-standard-analyzer.html) anyways lower-cases everything

The result of this is that we probably need both in some cases, e.g. author names (actually that would solve a big problem of trying to find all records by `Panero, Pablo`, without also getting results where [`Panero, Jose`, `Garcia, Pablo`] are authors).

File: zenodo/modules/deposit/mappings/v7/deposits/records/record-v1.0.0.json [hunk 1]
```diff
@@ -222,22 +222,22 @@
         "type": "date"
       },
       "title": {
-        "type": "text"
+        "type": "keyword"
```
Comment (slint) [inline zenodo/modules/deposit/mappings/v7/deposits/records/record-v1.0.0.json:225]: Based on https://github.com/zenodo/zenodo/pull/2446#issuecomment-1479311413, I would say we keep this as `text`

File: zenodo/modules/deposit/mappings/v7/deposits/records/record-v1.0.0.json [hunk 2]
```diff
@@ -359,7 +359,7 @@
       "references": {
         "properties": {
           "raw_reference": {
-            "type": "text"
+            "type": "keyword"
```
Comment (slint) [inline zenodo/modules/deposit/mappings/v7/deposits/records/record-v1.0.0.json:362]: Keep `text`, and apply the same `.keyword` trick as above.

File: zenodo/modules/deposit/mappings/v7/deposits/records/record-v1.0.0.json [hunk 3]
```diff
@@ -371,7 +371,7 @@
             "type": "keyword"
           },
           "title": {
-            "type": "text"
+            "type": "keyword"
```
Comment (slint) [inline zenodo/modules/deposit/mappings/v7/deposits/records/record-v1.0.0.json:374]: Same `text` + `fields.keyword` trick.

**Review by slint — COMMENTED** [review]
Some reconsiderations for existing fields, and applying the `fields.keyword` trick for others. Afterwards, we're good and should be able to merge.

**Pull request merged**